editoast/src/views/train_schedule_set.rs

```diff
@@ -0,0 +1,65 @@
+use database::DbConnectionPoolV2;
+use utoipa::ToSchema;
+
+use crate::error::Result;
+use crate::models::train_schedule_set::TrainScheduleSet;
+use crate::views::AuthenticationExt;
+use crate::views::AuthorizationError;
+use axum::Extension;
+use axum::extract::Json;
+use axum::extract::State;
+use serde::Deserialize;
+use serde::Serialize;
+use std::sync::Arc;
+
+#[derive(Serialize, Deserialize, ToSchema)]
+pub(in crate::views) struct TrainScheduleSetResponse {
+    #[serde(flatten)]
+    train_schedule_set: TrainScheduleSet,
+    train_schedule_count: u64,
+}
+
+#[derive(Serialize, Deserialize, ToSchema)]
+pub(in crate::views) struct TrainScheduleSetCreateForm {
+    catalogue_entry_id: Option<i64>,
```

Comment:
```suggestion
```

?

Reply:
@younesschrifi is right a train schedule set can be created without cantalogue_entry.

However, the backend should check (in DB), and probably in the view, that when a catalogue entry is not provided the "train schedule set" can not be published.